[BUG][0.11.3] `panic: abspath "/opt/actions-runner/_work/a/b/c" does not start with root "/opt/actions-runner/_work/a/a"`

**Describe the bug**

After updating the version from 0.11.2 to 0.11.3 , 

```shell
terramate create --all-terragrunt --tags dev
```

command started to fail with the weird error:

```log
panic: abspath "/opt/actions-runner/_work/a/b/c" does not start with root "/opt/actions-runner/_work/a/a"

goroutine 1 [running]:
github.com/terramate-io/terramate/project.PrjAbsPath({0xc000b0c2d0, 0x2d}, {0xc001aaf480, 0x39})
	/home/runner/work/terramate/terramate/project/project.go:126 +0x1b9
github.com/terramate-io/terramate/tg.ScanModules({0xc000b0c2d0, 0x2d}, {{0xc000178aed?, 0x3c?}}, 0x1)
	/home/runner/work/terramate/terramate/tg/tg_module.go:183 +0x2031
github.com/terramate-io/terramate/cmd/terramate/cli.(*cli).initTerragrunt(0xc000c9a000)
	/home/runner/work/terramate/terramate/cmd/terramate/cli/cli.go:1454 +0x73
github.com/terramate-io/terramate/cmd/terramate/cli.(*cli).scanCreate(0xc000c9a000)
	/home/runner/work/terramate/terramate/cmd/terramate/cli/cli.go:1447 +0x2c5
github.com/terramate-io/terramate/cmd/terramate/cli.(*cli).run(0xc000c9a000)
	/home/runner/work/terramate/terramate/cmd/terramate/cli/cli.go:706 +0x4f3
github.com/terramate-io/terramate/cmd/terramate/cli.Exec({0x19ac959, 0x6}, {0xc000142010, 0x4, 0x4}, {0x200a8a0, 0xc00012c040}, {0x200a8c0, 0xc00012c048}, {0x200a8c0, ...})
	/home/runner/work/terramate/terramate/cmd/terramate/cli/cli.go:388 +0xfc
main.main()
	/home/runner/work/terramate/terramate/cmd/terramate/main.go:20 +0x89
```

I'm wondering why the root is set to `/opt/actions-runner/_work/a/a` (The directory `a` is repeated)


**Expected behavior**

don't fail.



**Environment (please complete the following information):**

```shell
Operating System
  Distribution: Ubuntu 22.04.5 LTS
  Architecture: x86_64
```

Hi @i4ki, we are working on the same project at the moment as coworkers.

We are not using symlinks in the project. We are just using git sources to define our terragrunt stacks terraform source. 


> Are you using terramate -C <dir> ... ?

No, we use terramate create --all-terragrunt



> does it also happen when you run `terramate create --all-terragrunt` locally in your machine?
Yes, this is the error trace:

```
panic: abspath "/Users/myuser/reponame" does not start with root "/Users/myuser/reponame/"

goroutine 1 [running]:
github.com/terramate-io/terramate/project.PrjAbsPath({0x14000231280, 0x31}, {0x1400090adc0, 0x3d})
        /home/runner/work/terramate/terramate/project/project.go:126 +0x1d8
github.com/terramate-io/terramate/tg.ScanModules({0x14000231280, 0x31}, {{0x140002309b1?, 0x40?}}, 0x1)
        /home/runner/work/terramate/terramate/tg/tg_module.go:183 +0x19ec
github.com/terramate-io/terramate/cmd/terramate/cli.(*cli).initTerragrunt(0x14000c10400)
        /home/runner/work/terramate/terramate/cmd/terramate/cli/cli.go:1454 +0x60
github.com/terramate-io/terramate/cmd/terramate/cli.(*cli).scanCreate(0x14000c10400)
        /home/runner/work/terramate/terramate/cmd/terramate/cli/cli.go:1447 +0x2cc
github.com/terramate-io/terramate/cmd/terramate/cli.(*cli).run(0x14000c10400)
        /home/runner/work/terramate/terramate/cmd/terramate/cli/cli.go:706 +0x4dc
github.com/terramate-io/terramate/cmd/terramate/cli.Exec({0x103d9a003, 0x6}, {0x1400003a0d0, 0x2, 0x2}, {0x104869fa8, 0x1400007a008}, {0x104869fc8, 0x1400007a010}, {0x104869fc8, ...})
        /home/runner/work/terramate/terramate/cmd/terramate/cli/cli.go:388 +0xb0
main.main()
        /home/runner/work/terramate/terramate/cmd/terramate/main.go:20 +0x94
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -34,6 +34,8 @@ Given a version number `MAJOR.MINOR.PATCH`, we increment the:
 
 - Fix the command-line parsing of `run` and `script run` which were not failing from unknown flags.
 - Fix `create --all-terragrunt` creating Terragrunt stacks with cycles.
+- Panic in the Terragrunt integration when the project had modules with dependency paths outside the current Terramate project.
+  - Now Terramate throw a warning for such configurations.
 
 ## v0.11.3
 

--- a/cmd/terramate/cli/cli.go
+++ b/cmd/terramate/cli/cli.go
@@ -1609,7 +1609,7 @@ func (c *cli) initTerragrunt() {
 		for _, otherMod := range mod.After.Strings() {
 			// Parent stack modules must be excluded because of implicit filesystem ordering.
 			// Parent stacks are always executed before child stacks.
-			if mod.Path.HasPrefix(otherMod + "/") {
+			if otherMod == "/" || mod.Path.HasPrefix(otherMod+"/") {
 				continue
 			}
 			// after stacks must not be defined as child stacks

--- a/e2etests/core/create_all_terragrunt_test.go
+++ b/e2etests/core/create_all_terragrunt_test.go
@@ -438,6 +438,68 @@ func TestCreateAllTerragrunt(t *testing.T) {
 				StderrRegex: "Found a dependency cycle between modules",
 			},
 		},
+		{
+			name: "Terragrunt with dependency defined outside of the Terramate project",
+			layout: []string{
+				hclfile("terragrunt.hcl", Doc(
+					Block("terraform",
+						Str("source", "github.com/some/repo"),
+					),
+					Block("dependency",
+						Labels("module2"),
+						Str("config_path", `../other`),
+					),
+					Block("dependencies",
+						Expr("paths", `["../other"]`),
+					))),
+				hclfile("../other/terragrunt.hcl", Doc(
+					Block("terraform",
+						Str("source", "github.com/some/repo"),
+					),
+				)),
+			},
+			want: RunExpected{
+				StderrRegexes: []string{
+					regexp.QuoteMeta("Warning: Dependency outside of Terramate project detected in `dependency.config_path` configuration. Ignoring."),
+					regexp.QuoteMeta("Warning: Dependency outside of Terramate project detected in `dependencies.paths` configuration. Ignoring."),
+				},
+				Stdout: nljoin(
+					"Created stack /",
+				),
+			},
+			wantOrder: []string{"."},
+		},
+		{
+			name: "Terragrunt with dependency defined outside of the Terramate project (sharing same base path)",
+			layout: []string{
+				hclfile("terragrunt.hcl", Doc(
+					Block("terraform",
+						Str("source", "github.com/some/repo"),
+					),
+					Block("dependency",
+						Labels("module2"),
+						Str("config_path", `../sandboxy`),
+					),
+					Block("dependencies",
+						Expr("paths", `["../sandboxy"]`),
+					))),
+				hclfile("../sandboxy/terragrunt.hcl", Doc(
+					Block("terraform",
+						Str("source", "github.com/some/repo"),
+					),
+				)),
+			},
+			want: RunExpected{
+				StderrRegexes: []string{
+					regexp.QuoteMeta("Warning: Dependency outside of Terramate project detected in `dependency.config_path` configuration. Ignoring."),
+					regexp.QuoteMeta("Warning: Dependency outside of Terramate project detected in `dependencies.paths` configuration. Ignoring."),
+				},
+				Stdout: nljoin(
+					"Created stack /",
+				),
+			},
+			wantOrder: []string{"."},
+		},
 	} {
 		tc := tc
 		t.Run(tc.name, func(t *testing.T) {

--- a/project/project.go
+++ b/project/project.go
@@ -122,8 +122,14 @@ func PrjAbsPath(root, abspath string) Path {
 	if !filepath.IsAbs(abspath) {
 		panic(fmt.Errorf("path %q is not absolute", abspath))
 	}
+	if root == abspath {
+		return NewPath("/")
+	}
+	if !strings.HasSuffix(root, string(filepath.Separator)) { // accounts for root=/
+		root = root + string(filepath.Separator)
+	}
 	if !strings.HasPrefix(abspath, root) {
-		panic(fmt.Errorf("abspath %q does not start with root %q", abspath, root))
+		panic(fmt.Errorf(`abspath %q does not start with root %q`, abspath, root))
 	}
 	d := filepath.ToSlash(strings.TrimPrefix(abspath, root))
 	if d == "" {

--- a/tg/tg_module.go
+++ b/tg/tg_module.go
@@ -5,11 +5,12 @@ package tg
 
 import (
 	"context"
+	"fmt"
 	"io"
 	"os"
-	"path"
 	"path/filepath"
 	"sort"
+	"strings"
 
 	"github.com/gruntwork-io/go-commons/env"
 	"github.com/gruntwork-io/terragrunt/config"
@@ -18,6 +19,7 @@ import (
 	"github.com/gruntwork-io/terragrunt/util"
 	"github.com/rs/zerolog/log"
 	"github.com/terramate-io/terramate/errors"
+	"github.com/terramate-io/terramate/printer"
 	"github.com/terramate-io/terramate/project"
 	"github.com/zclconf/go-cty/cty/function"
 )
@@ -180,6 +182,11 @@ func ScanModules(rootdir string, dir project.Path, trackDependencies bool) (Modu
 				if !filepath.IsAbs(depPath) {
 					depPath = filepath.Join(tgMod.Path, depPath)
 				}
+				if depPath != rootdir && !strings.HasPrefix(depPath, rootdir+string(filepath.Separator)) {
+					warnDependencyOutsideProject(mod, depPath, "dependency.config_path")
+
+					continue
+				}
 				dependsOn[project.PrjAbsPath(rootdir, depPath)] = struct{}{}
 			}
 		}
@@ -191,14 +198,17 @@ func ScanModules(rootdir string, dir project.Path, trackDependencies bool) (Modu
 		}
 
 		if tgConfig.Dependencies != nil {
-			for _, dep := range tgConfig.Dependencies.Paths {
-				var p project.Path
-				if !path.IsAbs(dep) {
-					p = mod.Path.Join(dep)
-				} else {
-					logger.Debug().Str("dep-path", dep).Msg("ignore absolute path")
+			for _, depPath := range tgConfig.Dependencies.Paths {
+				if !filepath.IsAbs(depPath) {
+					depPath = filepath.Join(tgMod.Path, depPath)
+				}
+				if depPath != rootdir && !strings.HasPrefix(depPath, rootdir+string(filepath.Separator)) {
+					warnDependencyOutsideProject(mod, depPath, "dependencies.paths")
+
 					continue
 				}
+
+				p := project.PrjAbsPath(rootdir, depPath)
 				mod.After = append(mod.After, p)
 			}
 			mod.After.Sort()
@@ -251,4 +261,12 @@ func newTerragruntOptions(dir string) *options.TerragruntOptions {
 	opts.OriginalIAMRoleOptions = opts.IAMRoleOptions
 
 	return opts
+}
+
+func warnDependencyOutsideProject(mod *Module, dep string, field string) {
+	printer.Stderr.WarnWithDetails(fmt.Sprintf("Dependency outside of Terramate project detected in `%s` configuration. Ignoring.", field),
+		errors.E("The Terragrunt module %s depends on the module at %s, which is located outside of the your current"+
+			" Terramate project. To resolve this ensure the dependent module is part of your Terramate project.",
+			mod.Path, dep),
+	)
 }

